integrations/aws/main.py

```diff
@@ -69,19 +73,37 @@ async def _handle_global_resource_resync(
 async def resync_all(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
     if kind in iter(ResourceKindsWithSpecialHandling):
         return
+
     await update_available_access_credentials()
     is_global = is_global_resource(kind)
-    async for credentials in get_accounts():
-        if is_global:
-            async for batch in _handle_global_resource_resync(kind, credentials):
-                yield batch
-        else:
-            async for session in credentials.create_session_for_each_region():
-                try:
-                    async for batch in resync_cloudcontrol(kind, session):
-                        yield batch
-                except Exception:
-                    continue
+    semaphore = asyncio.BoundedSemaphore(MAX_CONCURRENT_TASKS)
+
+    async def handle_account(
+        credentials: AwsCredentials,
+    ) -> ASYNC_GENERATOR_RESYNC_TYPE:
+        """Function to handle fetching resources for a single account."""
+        async with semaphore:
+            errors, regions = [], []
+            if is_global:
+                async for batch in _handle_global_resource_resync(kind, credentials):
+                    yield batch
+            else:
+                async for session in credentials.create_session_for_each_region():
+                    try:
+                        async for batch in resync_cloudcontrol(kind, session):
+                            yield batch
+                    except Exception as exc:
+                        regions.append(session.region_name)
+                        errors.append(exc)
+                        continue
+            if errors:
+                message = f"Failed to fetch {kind} for these regions {regions} with {len(errors)} errors in account {credentials.account_id}"
+                raise ExceptionGroup(message, errors)
+
+    tasks = [handle_account(credentials) async for credentials in get_accounts()]
+    if tasks:
+        async for batch in stream_async_iterators_tasks(*tasks):
```

Comment:
Let's batch processes together, I think there's no limitaiton here (i.e. what if we have 20,000 accounts?)